lib/ViewScopeRector.php

```diff
@@ -47,13 +61,32 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
+        $inferredType = $this->inferTypeFromController("\IndexController", $node->exprs[0]->name);
+        if (!$inferredType) {
+            return null;
+        }
+
         // https://github.com/rectorphp/rector/blob/main/docs/how_to_work_with_doc_block_and_comments.md
         $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
-        $phpDocInfo->addTagValueNode(new VarTagValueNode(new IdentifierTypeNode('string'), '$'. $node->exprs[0]->name, ''));
+        $phpDocInfo->addTagValueNode(new VarTagValueNode($inferredType, '$' . $node->exprs[0]->name, ''));
 
         return $node;
     }
 
+    /**
+     * @param class-string $controllerClass
+     * @param string $propertyName
+     * @throws \PHPStan\Reflection\MissingPropertyFromReflectionException
+     */
+    private function inferTypeFromController($controllerClass, $propertyName): TypeNode
+    {
+        $classReflection = $this->reflectionProvider->getClass($controllerClass);
+        // XXX where to get the $scope from?
+        $propertyReflection = $classReflection->getProperty($propertyName);
```

Comment:
at this point, rector is analysing a [simplistic view file](https://github.com/staabm/rector-view-scope/blob/main/tests/Fixture/index_home.php.inc) and I want to lookup the corresponding controller for the view beeing analysed (`$controllerClass` - hardcoded for simplicity atm).

I want to lookup the controllers' properties, so I can infer types of variables based from the controller. which view belongs to which controller is a implementation detail from the framework which will be used in the end.

I know the path where the controller file will reside and want to use static reflection for this controller-class.